### PR TITLE
[70-74] `GraphStageLogicSpec`

### PR DESCRIPTION
**Pull request by eaba**
## Changes

- Change `AssertAllStagesStopped` to `AssertAllStagesStoppedAsync`

### CODE REVIEW

File: src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs [hunk 1]
```diff
@@ -273,37 +274,36 @@ public GraphStageLogicSpec(ITestOutputHelper output) : base(output, Config)
...
-                    .Via(new ReadNEmitN(6))
-                    .RunWith(this.SinkProbe<int>(), Materializer)
-                    .Request(10)
-                    .ExpectComplete();
+            await this.AssertAllStagesStoppedAsync(() => {
+                Source.From(Enumerable.Range(1, 5))                                                                             
+                .Via(new ReadNEmitN(6))                                                                             
+                .RunWith(this.SinkProbe<int>(), Materializer)                                                                             
+                .Request(10)                                                                             
+                .ExpectComplete();
```
Comment (Aaronontheweb) [inline src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs]: Is this `await`able?

**Review by Aaronontheweb — COMMENTED** [review]
Left some comments

File: src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs [hunk 2]
```diff
@@ -276,27 +276,25 @@ public GraphStageLogicSpec(ITestOutputHelper output) : base(output, Config)
...
-                Source.From(Enumerable.Range(1, 10))                                                                         
+            await this.AssertAllStagesStoppedAsync(async() => {
+                await Source.From(Enumerable.Range(1, 10))                                                                         
                 .Via(new ReadNEmitN(2))                                                                         
                 .RunWith(this.SinkProbe<int>(), Materializer)                                                                         
                 .Request(10)                                                                         
                 .ExpectNext(1, 2)                                                                         
-                .ExpectComplete();
-                return Task.CompletedTask;
+                .ExpectCompleteAsync();
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs:285]: @Aaronontheweb

File: src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs [hunk 3]
```diff
@@ -276,27 +276,25 @@ public GraphStageLogicSpec(ITestOutputHelper output) : base(output, Config)
...
-            await this.AssertAllStagesStoppedAsync(() => {
-                Source.From(Enumerable.Range(1, 5))                                                                             
+            await this.AssertAllStagesStoppedAsync(async() => {
+                await Source.From(Enumerable.Range(1, 5))                                                                             
                 .Via(new ReadNEmitN(6))                                                                             
                 .RunWith(this.SinkProbe<int>(), Materializer)                                                                             
                 .Request(10)                                                                             
-                .ExpectComplete();
-                return Task.CompletedTask;
+                .ExpectCompleteAsync();
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs:297]: good? @Aaronontheweb

File: src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs (hunk 2 above)
Comment (Aaronontheweb) [reply src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs:285]: you have a bunch of PRs open where you'll need to do this @eaba

**Review by Aaronontheweb — APPROVED** [review]
LGTM

File: src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs (hunk 3 above)
Comment (Aaronontheweb) [reply src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs:297]: Looks good, but you have a lot of code in other PRs today that looks like:

```csharp
...                        
  .ExpectComplete();
  return Task.CompletedTask;
```

You should change that to 

```csharp
.ExpectCompleteAsync();
```

Comment (eaba) [reply src/core/Akka.Streams.Tests/Implementation/GraphStageLogicSpec.cs:297]: Yeah, I will do it - bit by bit!

**Pull request merged**